lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -777,6 +856,7 @@ struct circt::firrtl::detail::BundleTypeStorage : mlir::TypeStorage {
     RecursiveTypeProperties props{true, false, false};
     uint64_t fieldID = 0;
     fieldIDs.reserve(elements.size());
+    isConst = !elements.empty();
```

Comment:
`BundleType` currently implicitly determines its own constness based on the constness of its elements. This means that an empty bundle can only be one of const or non-const (I chose non-const). I wasn't sure if it was worth being able to differentiate the two for now.